sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -235,7 +235,7 @@ case class AlterTableSetPropertiesCommand(
     // direct property.
     val newTable = table.copy(
       properties = table.properties ++ properties,
-      comment = properties.get("comment"))
+      comment = properties.get("comment").orElse(table.comment))
```

Comment:
This looks fine. However, when setting properties for Hive table, a `None` comment will be ignored and won't override the comment.

We don't do similar thing in `InMemoryCatalog` and just override it with the `None` comment.

Not sure if we should fix it here or in `InMemoryCatalog`.